### ISSUE TITLE
./build.sh cann`t find uhd

### ISSUE THREAD

**Issue by Ari100tell** [RangeNetworks/dev#30]
I try install OpenBTS (and otherr software) on Ubuntu 15.04 (Vivid), but after running installation I get next messages in terminal:

-# checking build dependencies
dpkg-query: package 'uhd' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
-# - missing uhd, installing dependency
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package uhd
-# - done

After installation I cann`t run program servises and while I try running ./OpenBTS --- I get:

1431196263.361118 140097295300416: 
Starting the system...
ALERT 22639:22639 2015-05-09T21:31:03.3 TRXManager.cpp:492:setTSC: SETTSC failed with status -1
ALERT 22639:22639 2015-05-09T21:31:03.3 TRXManager.cpp:456:powerOn: POWERON failed with status -22
ALERT 22639:22639 2015-05-09T21:31:03.3 TRXManager.cpp:565:clearHandover: NOHANDOVER(0) failed with status -1
ALERT 22639:22639 2015-05-09T21:31:03.3 GSML1FEC.cpp:578:handoverPending: SCH handover clear failed
ALERT 22639:22639 2015-05-09T21:31:03.3 TRXManager.cpp:565:clearHandover: NOHANDOVER(0) failed with status -1
ALERT 22639:22639 2015-05-09T21:31:03.3 GSML1FEC.cpp:578:handoverPending: FCCH handover clear failed
ALERT 22639:22639 2015-05-09T21:31:03.3 TRXManager.cpp:565:clearHandover: NOHANDOVER(0) failed with status -1
ALERT 22639:22639 2015-05-09T21:31:03.3 GSML1FEC.cpp:578:handoverPending: BCCH handover clear failed
..........................................

**Comment by dnng** [comment]
At the build.sh script, at line 114, you have:

```bash
sudo bash -c 'echo "deb http://files.ettus.com/binaries/uhd_stable/repo/uhd/ubuntu/`lsb_release -cs` `lsb_release -cs` main" > /etc/apt/sources.list.d/ettus.list'
```

If you follow that link, you'll see that the most recent binary over there is for the Ubuntu Trusty, which is Ubuntu 14.04. That is why you can't find it for 15.04.

You will have to download the UHD repo and build it manually for your machine. Be sure to use the proper version (I believe that for Ubuntu 14.04 the version is uhd_003.008.000.

**Comment by Ari100tell** [comment]
I am build UHD manually(configure-make-make install) before my first post, but build.sh script didn`t watch UHD-library.

**Comment by AprilianiHutagaol** [comment]
I try install OpenBTS on Debian Wheezy, but after running installation I get next messages in terminal:

-# checking build dependencies
dpkg-query: package 'uhd' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
-# - missing uhd, installing dependency
Reading package lists... Done
Building dependency tree

Reading state information... Done
E: Unable to locate package uhd
-# - done

Please help, thanks a lot

**Comment by Basterlord** [comment]
The problem is that the uhd packet is not installed there is one with the same name in the repository just do this for Debian like Linux system: 

sudo apt-get update && sudo apt-get install uhd

To see all the packets relative to that name use:

sudo apt-cache search uhd

Review the list to see all the packets similar to that name.

**Comment by JM1800** [comment]
Hi, I'm having the same problem but I am using Ubuntu server 12.04 LTS so the uhd package should be recognised I guess. Help please.

**Comment by mgkuhn** [comment]
build.sh uses an out-of-date source (http://files.ettus.com/binaries/uhd_stable/repo/uhd/ubuntu/) for the Ettus-compiled Ubuntu uhd binaries, which Ettus has not updated since 2014. They are now using launchpad.net instead: https://launchpad.net/~ettusresearch/+archive/ubuntu/uhd

The current official way to install Ettus' Ubuntu uhd binaries is documented at http://files.ettus.com/manual/page_install.html#install_linux_pkgmgr namely

```
sudo add-apt-repository ppa:ettusresearch/uhd
sudo apt-get update
sudo apt-get install uhd
```

**Comment by iedemam** [comment]
fixed in latest push for 16.04 compat
